### ISSUE TITLE
UI suggestions 🖍️

### ISSUE THREAD

**Issue by janjagusch** [janjagusch/easy-way#9]

**Comment by janjagusch** [comment]
Okay, so the first thing that we need to agree on is what is the scope of the app. Because, according to the project description, it should have two features:


1. > The quick navigation should make it possible to quickly find the article you are looking for by simply entering an article (text and/or speech) and to display and navigate the way to it from your current location.

2. > On the other hand, with the optimal route, the shopping basket can be filled before shopping. So that an optimal route through the store can be created and the app can display your current location and the route which you have to follow for the shopping basket.

I would suggest to focus on one feature first. I would say that we should start with 1. What do you think? @Hossam47 and @kiranvasudev?

**Comment by hossamwaziry** [comment]
1- we can get the current location of user by the nearest QR Code and build the path depending on the distance between those two points

**Comment by janjagusch** [comment]
For feature 1, I have a certain idea of how to app flow could look like:

1. Start screen: User is instructed to scan QR code.
![image](https://user-images.githubusercontent.com/25852654/96332336-b2a69380-1063-11eb-964a-9bc9ddd3f46f.png)
2. Supermarket overview screen: Layout of supermarket and current location (retrieved from QR code) is displayed. User has button to find an item in the supermarket (which triggers screen 3).
![image](https://user-images.githubusercontent.com/25852654/96332367-e08bd800-1063-11eb-8a61-41858e79375b.png)
3. Article selection view: User can search and select article they are looking for. User can type in the textbox and suggestions are displayed below. Selecting an item triggers screen 4.
![image](https://user-images.githubusercontent.com/25852654/96332398-07e2a500-1064-11eb-8c35-5820f508f61c.png)
4. Shortest path view: Location of item and shortest path to item are displayed on navigation view.
![image](https://user-images.githubusercontent.com/25852654/96332436-3bbdca80-1064-11eb-9e04-9c68783eed57.png)

**Comment by hossamwaziry** [comment]
In screen number 2 : add item will be Textfield ?

**Comment by janjagusch** [comment]
> In screen number 2 : add item will be Textfield ?

addItem would be a button that brings you to screen 3.

And maybe it should be called "searchItem" because you can only search for **one** item.

**Comment by hossamwaziry** [comment]
Okay

**Comment by kiranvasudev** [comment]
>I would suggest to focus on one feature first. I would say that we should start with 1. 

I would suggest that we also focus on 2. I feel like it is a more realistic use case.

**Comment by janjagusch** [comment]
> > I would suggest to focus on one feature first. I would say that we should start with 1.
> 
> I would suggest that we also focus on 2. I feel like it is a more realistic use case.

Yes, it is definitely the more exciting use case! I just found it a lot harder to make it work on the app (UI wise). We have it working in the backend already though, so I was thinking about showing it inside a Jupyter Notebook (we could even deploy it on binder and make it available to everyone). What do you think @kiranvasudev?

**Comment by kiranvasudev** [comment]
>I just found it a lot harder to make it work on the app (UI wise).

Why do you say this? If we can get the path to one item(in the UI), it should be more or less similar for more items right?

@Hossam47 could you give us some input to this?

**Comment by janjagusch** [comment]
> > I just found it a lot harder to make it work on the app (UI wise).
> 
> Why do you say this? If we can get the path to one item(in the UI), it should be more or less similar for more items right?
> 

I guess one thing I found hard is to show which item to collect at which section in the supermarket. You would need some sort of hover for that.

Besides the app flow would be a bit different, because you wouldn't start with a QR code. They said you would use this feature before going to the supermarket. So I guess the first thing in the app flow would be to select the supermarket you want to go to. Then instead of just selecting one item, you could select multiple. I would be okay with drawing some mockups for that as well, but it really depends on what @Hossam47 thinks and how much time he has. :)

**Comment by kiranvasudev** [comment]
>So I guess the first thing in the app flow would be to select the supermarket you want to go to. Then instead of just selecting one item, you could select multiple.

I was thinking the same. And to me, instead of focusing on just one product and getting a route for it, we should do a multiple item path navigator. Because, realistically, I would say that a person would calculate the path to a single item if he/she is a tourist/person(?) who goes to a new supermarket.
For the regular visitors, they would mostly just use Feature 2.

>I would be okay with drawing some mockups for that as well, but it really depends on what @Hossam47 thinks and how much time he has. :)

Agree with this. I have no idea how difficult this would be to build on the app.

**Comment by hossamwaziry** [comment]
@kiranvasudev @janjagusch it will take a long time , i think we can work on the first feature and planing to the second
